client/src/legacy/LeftAndMain.js

```diff
@@ -928,46 +923,14 @@ $.entwine('ss', function($) {
      *   Used to mark a specific tab as active regardless of the previously saved options.
      */
     restoreTabState: function(overrideStates) {
-      var self = this, url = this._tabStateUrl(),
-        hasSessionStorage = (typeof(window.sessionStorage)!=="undefined" && window.sessionStorage),
-        sessionData = hasSessionStorage ? window.sessionStorage.getItem('tabs-' + url) : null,
-        sessionStates = sessionData ? JSON.parse(sessionData) : false;
-
-      var tabset = this.find('.cms-tabset, .ss-tabset');
-
-      if (tabset.length) {
-        tabset.each(function() {
-          var index, tab,
-            tabset = $(this),
-            tabsetId = tabset.attr('id'),
-            forcedTab = tabset.children('ul').children('li.ss-tabs-force-active');
-
-          if(!tabset.data('tabs')){
-            return; // don't act on uninit'ed controls
-          }
-
-          // The tabs may have changed, notify the widget that it should update its internal state.
-          tabset.tabs('refresh');
-
-          // Make sure the intended tab is selected. Only force the tab on the correct tabset though
-          if(forcedTab.length) {
-            index = forcedTab.first().index();
-          } else if(overrideStates && overrideStates[tabsetId]) {
-            tab = tabset.find(overrideStates[tabsetId].tabSelector);
-            if(tab.length){
-              index = tab.index();
-            }
-          } else if(sessionStates) {
-            $.each(sessionStates, function(i, state) {
-              if(tabsetId == state.id) {
-                index = state.selected;
-              }
-            });
-          }
-          if(index !== null){
-            tabset.tabs('option', 'active', index);
-            self.trigger('tabstaterestored');
-          }
+      const tabsets = this.find('.cms-tabset, .ss-tabset');
+
+      if (tabsets.length) {
+        tabsets.each(function() {
+          const tabset = $(this);
+          const tabsetId = tabset.attr('id');
+          const overrideState = (overrideStates && overrideStates[tabsetId]) ? overrideStates[tabsetId] : null;
+          tabset.restoreState(overrideState);
```

Comment:
The state restore functionality has moved to `TabSet.js` so that each tab can be responsible for restoring its own state.
This is necessary because `onAdd()` is now triggered asyncronously _some time after_ the element is added (see changes in entwine to using mutation observers) which means the tabs may not be initialised yet when this function is called.